scripts/pharmgkb/drug_gene_relations/pharm.py

```diff
@@ -0,0 +1,644 @@
+        pharm_to_chembl_dict[pharm_id]
+        if pharm_id in pharm_to_chembl_dict else ''
+        for pharm_id in drugs_df['PharmGKB Accession Id']
+    ]
+
+    # read in chembl based off of pubchem compound id
+    pubchem_chembl_df = pd.read_csv(
+        './conversion/pubchem_id_to_chembl_combined.csv')
+    pubchem_to_chembl_dict = pd.Series(
+        pubchem_chembl_df['ChEMBL ID'].values,
+        index=pubchem_chembl_df['PubChem ID']).to_dict()
+
+    drugs_df['Chembl2'] = [
+        pubchem_to_chembl_dict[pubchem_id]
+        if pubchem_id in pubchem_to_chembl_dict else ''
+        for pubchem_id in drugs_df['PubChem Compound Identifiers']
+    ]
+
+    # read in chembl ids based off of inchi id
+    inchi_keys = pd.read_csv(
+        './conversion/inchi_to_inchi_key_combined.csv')['InChI Key']
+    inchis = pd.read_csv(
+        './conversion/inchi_to_inchi_key_combined.csv')['InChI']
+    chembl_ids = pd.read_csv(
+        './conversion/inchi_key_to_chembl_combined.csv')['ChEMBL ID']
+
+    inchi_to_chembl_dict = pd.Series(chembl_ids.values, index=inchis).to_dict()
+
+    drugs_df['Chembl3'] = [
+        inchi_to_chembl_dict[inchi] if inchi in inchi_to_chembl_dict else ''
+        for inchi in drugs_df['InChI']
+    ]
+    drugs_df['InChI Key'] = inchi_keys
+
+    drugs_df['Chembl ID'] = drugs_df.apply(lambda row: merge_chembls(
+        row['Chembl1'], row['Chembl2'], row['Chembl3']),
+                                           axis=1)
+
+    return drugs_df
+
+
+def get_drugs_df():
+    """Returns a combined drugs data frame.
+
+    Returns:
+        Dataframe containg chemicals.tsv, drugs.tsv as well as Chembl ID and
+        InChI Key information retreived from ./conversion/*.csv files.
+    """
+    chemicals_df = pd.read_csv('./raw_data/chemicals/chemicals.tsv', sep='\t')
+    drugs_df = pd.read_csv('./raw_data/drugs/drugs.tsv', sep='\t')
+
+    combined_df = drugs_df.append(chemicals_df, ignore_index=True)
+    combined_df = combined_df.drop_duplicates()
+
+    drugs_df = append_chembls(combined_df)
+    drugs_df.fillna('', inplace=True)
+
+    return drugs_df
+
+
+def format_text_list(text_list):
+    """Creates a formatted version of the given string according to MCF standards.
+
+    This is used to format many columns from the drugs and genes dataframes like
+    PubChem Compound Identifiers or NCBI Gene ID.
+
+    Args:
+        text_list: A single string representing a comma separated list. Some of
+            the items are enlcosed by double quotes and some are not.
+
+    Returns:
+        A string that is mcf property text values list enclosed by double quotes
+        and comma separated.
+
+    Example:
+        input: ('test1,
+            "Carbanilic acid, M,N-dimethylthio-, O-2-naphthyl ester", "test2"')
+        return: ('"test1",
+             "Carbanilic acid, M,N-dimethylthio-, O-2-naphthyl ester", "test2"')
+    """
+    if not text_list:
+        return ''
+    formatted_str = ''
+    joining = ''
```

Comment:
What's the purpose of this joining variable? Is there a reason for this and handling the different cases seperately for the replacement of the "" below?